data/mods/gen7letsgo/scripts.ts

```diff
@@ -1,3 +1,15 @@
+function checkMegaForme(this: BattleActions, species: Species, forme: string) {
+	const baseSpecies = this.dex.species.get(species.baseSpecies);
+	const altForme = this.dex.species.get(`${baseSpecies.name}-${forme}`);
+	if (
+		altForme.exists && !this.battle.ruleTable.isBannedSpecies(altForme) &&
+		!this.battle.ruleTable.isBanned('pokemontag:mega')
+	) {
+		return altForme.name;
+	}
+	return null;
+}
```

Comment:
Is there a good reason for this to be here instead of under `actions:`? (The use of `call` looks a little awkward to me, but that's just personal preference. Alternatively, you could pass, say, `battle`, as an explicit parameter.)

Reply:
If I put it under actions I would have to put the function name in the `ModdedBattleActions` type. Looking at it now, yeah `call` is probably unnecessary, will change that to passing in `battle`.